src/client/mod.rs

```diff
@@ -290,464 +215,122 @@ impl Client {
         Ok(Client {
             flags: flags,
             cluster_id: cluster_id,
-            mmv_path: mmv_path,
-            wi: MMVWriterInfo::new()
+            mmv_path: mmv_path
         })
     }
+    
+    pub fn export(&self, metrics: &mut [&mut MMVWriter]) -> io::Result<()> {
```

Comment:
> This works only if there is a 1:1 relationship between a metric and a client. There might be an application creating 2 clients that might want to share a metric. In that case, if both export concurrently, I don't think both will be able to get a mutable reference to the same metric (from my understanding)

The compiler won't allow us to export the same metric concurrently, say in two threads with separate clients, without a mutex first wrapping it and each thread locking/unlocking it appropriately when exporting. In that case a data race will surely be avoided, but the metric will always write to last MMV it was exported to.